web/testing/promise-api/index.html

```diff
@@ -36,15 +38,41 @@
 
     <!-- Add keyboard management script for local selection of keyboards to use -->
     <script src="promisehdr.js"></script>
+    <script src="eventLogging.js"></script>
 
     <!-- Initialization: set paths to keyboards, resources and fonts as required -->
     <script>
       // Thank you to https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams
       const alertType = (new URLSearchParams(window.location.search)).get("useAlerts") != "false";
-      var alertText = alertType ? 'enabled(default)' : 'disabled';
 
       var kmw=window.keyman;
 
+      var loggingActive = false;
+
+      function eventLogChange() {
+        const logActiveRadio = document.getElementById('eventLog-on');
+        const offLabel = document.getElementById('event-label-off');
+        const onLabel = document.getElementById('event-label-on');
+
+        const loggingDiv = document.getElementById('logZone');
+        const hasLoggingText = !!document.getElementById('eventLogging').value;
+
+        const active = logActiveRadio.checked;
+        loggingActive = active;
+
+        offLabel.className = active ? '' : 'active';
+        onLabel.className  = active ? 'active' : '';
+
+        loggingDiv.style.display = active || hasLoggingText ? 'block' : 'none';
+      }
+
+      function flushLogs() {
+        const loggingElement = document.getElementById('eventLogging');
+        loggingElement.value = '';
+
+        eventLogChange();
+      }
```

Comment:
Can we put this new js in eventLogging.js? It seems to belong there?

Reply:
I thought about it, but decided against it because this one deals far more directly with the UI aspects, whereas those are about the event interception and logging aspects.  Like, both are completely page-UI focused.

Still, totally can move it if you'd prefer; it is admittedly more code inlined on the main page.

Reply:
I like the reorg. Less inline code is better.